gnucash/import-export/aqb/gnc-ab-utils.c

```diff
@@ -276,25 +228,30 @@ gnc_ab_get_ab_account(const AB_BANKING *api, Account *gnc_acc)
 
     if (account_uid > 0)
     {
-        ab_account = AB_Banking_GetAccount(api, account_uid);
+        gint rv;
+
+        rv = AB_Banking_GetAccountSpecByUniqueId(api, account_uid, &ab_account);
 
-        if (!ab_account && bankcode && *bankcode && accountid && *accountid)
+        if ( (rv<0 || !ab_account) && bankcode && *bankcode && accountid && *accountid)
         {
             g_message("gnc_ab_get_ab_account: No AB_ACCOUNT found for UID %d, "
                       "trying bank code\n", account_uid);
+#if 0 /* TODO */
```

Comment:
I find this and the following item a bit distressing. It implies that Martin intends to reimplement AB_Banking_GetAccountByCodeAndNumber(). This breaks the function for AQB6 and ISTM means that trying to support AQB6 for GC3.5 is perhaps misguided.

Reply:
I don't intend to remove account lookup by bank code and account id, I just wanted to get it done quickly to see whether it could be done and improve later (in case there was any interest in my changes, which didn't seem to be the case until now, so I didn't continue to work on it).

I chose to start with lookup by unique id because the combination of account number and bank code is very much outdated, at least in Europe, since there is only IBAN now. Banks seem to support account numbers for some time, but that will eventually go away.
Also, some banks in Germany chose to use the same bank code and account number for different accounts, so this combination is not unique with those banks.

Anyhow, in AqBanking6 there already is a function which can use several account attributes to find matching accounts (currently IBAN, BIC, account number, bank code, account type, currency etc.) which should be used here.

Reply:
Regards
Martin (AqBanking)

Reply:
Martin, 
Great, thanks. What's the function?